lib/puppet/provider/zabbix_proxy/ruby.rb

```diff
@@ -33,39 +23,178 @@ def self.prefetch(resources)
     end
   end
 
+  def self.proxies(proxyids = nil)
+    zbx.query(
+      method: 'proxy.get',
+      params: {
+        proxyids: proxyids,
+        output: 'extend',
+        selectInterface: %w[interfaceid type main ip port useip]
+      }.compact
+    )
+  end
+
+  # Convert from proxy hash returned by API into a resource properties hash
+  def self.proxy_properties_hash(p)
+    {
+      ensure: :present,
+      # Proxy object properties
+      proxyid: p['proxyid'].to_i,
+      name:    p['host'],
+      mode:    status_to_mode(p['status']),
+      # Proxy Interface object properties
+      interfaceid: p['interface'].is_a?(Hash) ? p['interface']['interfaceid'] : nil,
+      ipaddress:   p['interface'].is_a?(Hash) ? p['interface']['ip'] : nil,
+      use_ip:      if p['interface'].is_a?(Hash) then p['interface']['useip'] == '1' ? :true : :false end,
+      port:        p['interface'].is_a?(Hash) ? p['interface']['port'].to_i : nil
+    }
+  end
+
   def create
-    # Set some vars
-    host = @resource[:hostname]
-    ipaddress = @resource[:ipaddress]
-
-    # Normally 0 is active and 1 is passive, in the API, its 5 and 6
-    proxy_mode = @resource[:mode] + 5
-
-    use_ip = @resource[:use_ip]
-    port = @resource[:port]
-
-    # Check if we need to connect via ip or fqdn
-    use_ip = use_ip ? 1 : 0
-
-    zbx.proxies.create_or_update(
-      host: host,
-      status: proxy_mode,
-      interfaces: [
-        ip: ipaddress,
-        dns: host,
-        useip: use_ip,
-        port: port
-      ]
+    mode = @resource[:mode] || :active
+
+    if mode == :passive
+      useip = if @resource[:use_ip].nil?
+                1
+              else
+                @resource[:use_ip] == :true ? 1 : 0
+              end
+      interface = {
+        ip: @resource[:ipaddress] || '127.0.0.1',
+        dns: @resource[:hostname],
+        useip: useip,
+        port: @resource[:port] || 10_051
+      }
+    else
+      interface = nil
+    end
+
+    zbx.proxies.create(
+      {
+        host: @resource[:hostname],
+        status: self.class.mode_to_status(mode),
+        interface: interface
```

Comment:
haha :D I was about to open an issue for this. `interfaces` is deprecated as of Zabbix 2.2 in favor of `interface` and was removed in 3.4. Thank you! 👍

Reply:
My starting point was your code, so you can take credit for fixing this! :)